include/seqan3/alignment/pairwise/policy/affine_gap_init_policy.hpp

```diff
@@ -64,109 +64,103 @@ class affine_gap_init_policy
+     * \param[in]     cache        The cache storing hot helper variables.
      */
     template <typename cell_t, typename cache_t>
     constexpr auto init_origin_cell(cell_t && current_cell, cache_t & cache) const noexcept
     {
-        using std::get;
+        auto [score_cell, trace_cell] = current_cell;
 
-        auto & [main_score, hz_score, hz_trace] = get<0>(current_cell);
-        auto & prev_cell = get<0>(cache);
-        auto & vt_score = get<1>(prev_cell);
+        // Initialise the first cell.
+        score_cell.current = 0;
+        trace_cell.current = trace_directions::none;
 
-        main_score = 0;
-        get<2>(current_cell) = trace_directions::none; // store the trace direction
+        static_cast<derived_t const &>(*this).check_score(current_cell, cache);
 
         // Initialise the vertical matrix cell according to the traits settings.
         if constexpr (traits_type::free_second_leading_t::value)
         {
-            vt_score = 0;
-            get<2>(prev_cell) = trace_directions::none;  // cache vertical trace
+            score_cell.up = 0;
+            trace_cell.up = trace_directions::none;  // cache vertical trace
         }
-        else
+        else // Initialise with gap_open score
         {
-            vt_score = get<1>(cache);
-            get<2>(prev_cell) = trace_directions::up_open; // cache vertical trace
+            score_cell.up = cache.gap_open_score;
+            trace_cell.up = trace_directions::up_open; // cache vertical trace
         }
 
         // Initialise the horizontal matrix cell according to the traits settings.
         if constexpr (traits_type::free_first_leading_t::value)
         {
-            hz_score = 0;
-            hz_trace = trace_directions::none; // cache horizontal trace
+            score_cell.w_left = 0;
+            trace_cell.w_left = trace_directions::none; // cache horizontal trace
         }
-        else
+        else // Initialise with gap_open score
         {
-            hz_score = get<1>(cache);
-            hz_trace = trace_directions::left_open; // cache horizontal trace
+            score_cell.w_left = cache.gap_open_score;
+            trace_cell.w_left = trace_directions::left_open; // cache horizontal trace
         }
     }
 
     /*!\brief Initialises a cell in the first column of the dynamic programming matrix.
      * \tparam        cell_t       The underlying cell type.
      * \tparam        cache_t      The type of the cache.
      * \param[in,out] current_cell The current cell in the dynamic programming matrix.
-     * \param[in,out] cache        The cache storing hot helper variables.
+     * \param[in]     cache        The cache storing hot helper variables.
      */
     template <typename cell_t, typename cache_t>
     constexpr auto init_column_cell(cell_t && current_cell, cache_t & cache) const noexcept
     {
-        using std::get;
+        auto [score_cell, trace_cell] = current_cell;
```

Comment:
Don't you take a copy here instead of taking a reference? Do you want this?

Reply:
These are proxies anyway. So RVO should avoid any copying. But I've learned that RVO might not work with structured bindings. That is something that I need to re-investigate. But that would be a rather silent refactorisation in the future.

Reply:
Like, before doing further micro level optomisations first the things has to be thoroughly benchmarked. But that is something after vectorisation has added so we know the end state of it all.

Reply:
I don't see the `return` in the function? ... am I blind?

Reply:
The return type should be void. The score_cell is a proxy type that internally stores reference members to the corresponding cells within the matrix. So everything is lightweight anyway. The general hope is that the compiler can see this and directly uses the reference members eliminating all unnecessary copies. But of course I keep it in mind and benchmark it thoroughly. I will annotate what is going on.

Reply:
ah the tuple elements are references. 
If so it does not hurt to still make it `auto &` instead of `auto` for clarity. Godbolt doesn't show a difference. And I don't exactly know which copies you mean to avoid, but maybe you can explain this to me in person.

Reply:
I made it `const &`, which is semantically similar to taking it per copy.